### PR TITLE
Adds methods to VerbNet API to make it machine consumable through python data structures

### PR DESCRIPTION
**Pull request by chnsh**
This code makes VerbNet machine consumable as discussed in issue - #1721 

A VerbNet class comprises of members (lemmas), frames, thematic roles and subclasses
Features added:
1) Created methods to access frames, thematic roles, and subclasses individually.
2) All entities of verbnet are machine consumable are machine consumable now as python data structures
3) Added 'example' sentence of each frame in the frame entity.
3) Additionally, pretty printing code has been modified to now use these newly created methods.
4) Created tests for these features and added docstrings to these newly added methods.

Additionally, 
1) Refactored code and ensured uniformity in accessing verbnet entities using 'vnclass'
2) Few minor refactors

### CODE REVIEW

**Comment by nschneid** [comment]
This looks like it adds useful functionality. Does someone want to do a proper code review?

File: nltk/corpus/reader/verbnet.py [hunk 1]
```diff
@@ -172,9 +172,77 @@ def fileids(self, vnclass_ids=None):
...
+        3) Syntax
+        4) Semantics
+        
+        :param vnclass: A verbnet class identifier; or an ElementTree
+            containing the xml contents of a verbnet class.
+        :return: frames - a list of frame dictionaries
+        """
+        if isinstance(vnclass, string_types):
+            vnclass = self.vnclass(vnclass)
+        frames = list()
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: i would just use `[]`.

File: nltk/corpus/reader/verbnet.py [hunk 2]
```diff
@@ -172,9 +172,77 @@ def fileids(self, vnclass_ids=None):
...
+        vnframes = vnclass.findall('FRAMES/FRAME')
+        for vnframe in vnframes:
+            example_text = self._get_example_within_frame(vnframe)
+            description = self._get_description_within_frame(vnframe)
+            pos_syntax = self._get_syntactic_list_within_frame(vnframe)
+            semantics = self._get_semantics_within_frame(vnframe)
+            frames.append({
+                'example': example_text,
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: maybe it worth moving the function calls into the dict creation:
```python
frames.append({
    'example': self._get_example_within_frame(vnframe),
    ...
})

File: nltk/corpus/reader/verbnet.py [hunk 3]
```diff
@@ -172,9 +172,77 @@ def fileids(self, vnclass_ids=None):
             return [self._class_to_fileid[self.longid(vnclass_id)]
                     for vnclass_id in vnclass_ids]
 
+    def frames(self, vnclass):
+        """
+        Given a verbnet class, this method returns frames with members-
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: I would rephrase this to follow https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings
```python
def frames(self, vnclass):
    """Return frames with members.

    The members are:
    1) Example
    ...

"""

File: nltk/corpus/reader/verbnet.py [hunk 4]
```diff
@@ -172,9 +172,77 @@ def fileids(self, vnclass_ids=None):
...
+                'example': example_text,
+                'description': description,
+                'syntax': pos_syntax,
+                'semantics': semantics
+            })
+        return frames
+
+    def subclasses(self, vnclass):
+        """
+        Given a verbnet class, this method returns subclass ids (if they exist)
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: Docstring style.

File: nltk/corpus/reader/verbnet.py [hunk 5]
```diff
@@ -172,9 +172,77 @@ def fileids(self, vnclass_ids=None):
...
+        if isinstance(vnclass, string_types):
+            vnclass = self.vnclass(vnclass)
+
+        subclasses = [subclass.get('ID') for subclass in
+                      vnclass.findall('SUBCLASSES/VNSUBCLASS')]
+        return subclasses
+
+    def themroles(self, vnclass):
+        """
+        Given a verbnet class, this method returns thematic roles participating
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: Docstring format.

File: nltk/corpus/reader/verbnet.py [hunk 6]
```diff
@@ -251,15 +319,96 @@ def shortid(self, longid):
...
             raise ValueError('vnclass identifier %r not found' % longid)
 
     ######################################################################
-    #{ Pretty Printing
+    # { Frame access utility functions
+    ######################################################################
+
+    def _get_semantics_within_frame(self, vnframe):
+        """
+        A utility function to retrieve semantics within a frame in verbnet
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: Docstring format.

File: nltk/corpus/reader/verbnet.py [hunk 7]
```diff
@@ -70,30 +71,32 @@ def __init__(self, root, fileids, wrap_etree=False):
     """Regular expression used by ``_index()`` to quickly scan the corpus
        for basic information."""
 
-    def lemmas(self, classid=None):
+    def lemmas(self, vnclass=None):
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py:74]: You have tests, but it would be nice to have examples within the methods (lemmas, wordnetids).

**Review by dimazest — COMMENTED** [review]
Looks good to me, but I've never worked with the VerbNet API.

**Comment by chnsh** [comment]
Awesome. Thanks for the comments. I'll make the changes as soon as I have some time and I will push the commit

**Comment by alvations** [comment]
@chintanshah24 did you get a chance to make the changes suggested by @dimazest?

File: nltk/corpus/reader/verbnet.py (hunk 7 above)
Comment (chnsh) [reply nltk/corpus/reader/verbnet.py:74]: Can you elaborate on this? This is not really clear

**Comment by chnsh** [comment]
Worked on the comments and pushed a new commit

File: nltk/corpus/reader/verbnet.py (hunk 7 above)
Comment (dimazest) [reply nltk/corpus/reader/verbnet.py:74]: I mean the usage example in the format of docstrings. Something like https://github.com/nltk/nltk/blob/develop/nltk/parse/dependencygraph.py#L146

**Comment by alvations** [comment]
@chintanshah24 Thanks for the updates but it seems like edits from 7c419bfe10e3ceebbd4ae4c6ce826acb0cd02cdf has created typos from "VerbNet" -> "VernNet". Could you take another look at it and make the appropriate changes? Thank you!

**Comment by chnsh** [comment]
Fixed

File: nltk/corpus/reader/verbnet.py [hunk 8]
```diff
@@ -6,10 +6,10 @@
 # For license information, see LICENSE.TXT
 
 """
-An NLTK interface to the VerbNet verb lexicon
+An NLTK interface to the Verbnet verb lexicon
 
-For details about VerbNet see:
-http://verbs.colorado.edu/~mpalmer/projects/verbnet.html
+For details about Verbnet see:
+http://verbs.colorado.edu/~mpalmer/projects/Verbnet.html
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: it's http://verbs.colorado.edu/~mpalmer/projects/verbnet.html 

Verbnet.html doesn't work.

Also, it might worth leaving things as they were, that is to use 'VerbNet', not 'Verbnet'.

File: nltk/corpus/reader/verbnet.py [hunk 9]
```diff
@@ -21,18 +21,19 @@
...
+    From the Verbnet site: "Verbnet (VN) (Kipper-Schuler 2006) is the largest
     on-line verb lexicon currently available for English. It is a hierarchical
     domain-independent, broad-coverage verb lexicon with mappings to other
     lexical resources such as WordNet (Miller, 1990; Fellbaum, 1998), Xtag
     (XTAG Research Group, 2001), and FrameNet (Baker et al., 1998)."
 
-    For details about VerbNet see:
-    http://verbs.colorado.edu/~mpalmer/projects/verbnet.html
+    For details about Verbnet see:
+    http://verbs.colorado.edu/~mpalmer/projects/Verbnet.html
```
Comment (dimazest) [inline nltk/corpus/reader/verbnet.py]: it's http://verbs.colorado.edu/~mpalmer/projects/verbnet.html

**Review by dimazest — CHANGES_REQUESTED** [review]

File: nltk/corpus/reader/verbnet.py (hunk 8 above)
Comment (chnsh) [reply nltk/corpus/reader/verbnet.py]: Yes, I agree. I aplogize for the relentless renaming

File: nltk/corpus/reader/verbnet.py (hunk 9 above)
Comment (chnsh) [reply nltk/corpus/reader/verbnet.py]: Its done

**Comment by chnsh** [comment]
Any updates?

File: nltk/corpus/reader/verbnet.py (hunk 9 above)
Comment (dimazest) [reply nltk/corpus/reader/verbnet.py]: i think the link is still different from the original

**Review by dimazest — COMMENTED** [review]
I would keep VerbNet (capital V ad N) trough the file in the docstrings and comments.

**Comment by alvations** [comment]
@chintanshah24 could you do a git pull on your branch to update your copy of the develop branch of NLTK and the branch conflict due to the `authors.md` file should be resolved =)

**Comment by alvations** [comment]
@dimazest I've merged an edit to change the V/N capitals to VerbNet for the docstrings. 

Could you help to do another round of quick review and decide whether the PR is good to merge? 

Thanks in advance!

**Comment by alvations** [comment]
[CI:retest]

**Comment by alvations** [comment]
[CI: retest]

**Comment by dimazest** [comment]
Looks good to me

**Review by dimazest — APPROVED** [review]

**Comment by chnsh** [comment]
I'm a little confused. Is there anything left to do?

**Comment by alvations** [comment]
@chintanshah24 Nope, all's clear, looks good to merge now =) 

Sorry I took the liberty to change the docstring for the capital Ns and Vs as @dimazest requested and merged the conflict on the `authors.md`

Thanks @chintanshah24 for the PR! Thanks for reviewing it @dimazest !

**Comment by chnsh** [comment]
Please don't mention it. Thank you for making the requested changes. Have gotten too busy with work these days.
Thanks @alvations and @dimazest for this. My first open source contribution :)

**Comment by stevenbird** [comment]
Thanks @chintanshah24, @alvations, @dimazest
